src/core/reducers/autocomplete.js

```diff
@@ -0,0 +1,50 @@
+export const AUTOCOMPLETE_LOADED = 'core/autocomplete/AUTOCOMPLETE_LOADED';
+export const AUTOCOMPLETE_STARTED = 'core/autocomplete/AUTOCOMPLETE_STARTED';
+
+const initialState = {
+  suggestions: [],
+};
+
+export function autocompleteStart({ errorHandlerId, query }) {
+  if (!errorHandlerId) {
+    throw new Error('errorHandlerId is required');
+  }
+
+  if (!query) {
+    throw new Error('query is required');
+  }
+
+  return {
+    type: AUTOCOMPLETE_STARTED,
+    payload: { errorHandlerId, query },
+  };
+}
+
+export function autocompleteLoad({ results }) {
+  if (!results) {
+    throw new Error('results is required');
+  }
+
+  return {
+    type: AUTOCOMPLETE_LOADED,
+    payload: { results },
+  };
+}
+
+export default function reducer(state = initialState, action) {
+  const { payload } = action;
+  switch (action.type) {
+    case AUTOCOMPLETE_STARTED:
+      return state;
```

Comment:
I think resetting the suggestions would be good as I am often frustrated by autocompletes that leave old results on screen as I search. Perhaps we should just have a more robust loading screen. Instead of [returning `null` here](https://github.com/mozilla/addons-frontend/pull/2958/files#diff-c2d5de2068dbee8f3313dfa396c573b0R94) you could make the suggestions an array of `LoadingText` components as we do elsewhere in the app if there is a search happening.

I think if the query is deleted and the search hides/appears that's fine.

Let me know if that makes sense.